src/module/actor/sheets/SR5MatrixActorSheet.ts

```diff
@@ -93,6 +93,82 @@ export class SR5MatrixActorSheet extends SR5BaseActorSheet {
         html.find('.open-matrix-device').on('click', this._onOpenMatrixDevice.bind(this));
 
         html.find('.targets-refresh').on('click', this._onTargetsRefresh.bind(this));
+
+        html.find('.setup-pan').on('click', this._addAllEquippedWirelessDevicesToPAN.bind(this));
+
+        // Matrix Network
+        html.find('.connect-to-network').on('click', this._onConnectToMatrixNetwork.bind(this));
+        // Matrix Target - Connected Icons Visibility Switch
+        html.find('.toggle-connected-matrix-icons').on('click', this._onToggleConnectedMatrixIcons.bind(this));
+    }
+
+    /**
+     * Allow the user to select a matrix network to connect to.
+     */
+    async _onConnectToMatrixNetwork(event) {
+        event.stopPropagation();
+
+        const dialog = new SelectMatrixNetworkDialog(this.document);
+        const network = await dialog.select();
+        if (dialog.canceled) return;
+
+        await this.document.connectNetwork(network);
+        this.render();
+    }
+
+    /**
+     * Within the list of avialable matrix targets, toggle visibility of sub-sections for connected icons
+     * for a single matrix target.
+     *
+     * This is done by clicking on a specific icon by user input and will trigger:
+     * - switching out sheet display
+     * - provide a display of additional matrix icons underneath uuid
+     */
+    async _onToggleConnectedMatrixIcons(event) {
+        event.stopPropagation();
+
+        const uuid = Helpers.listItemUuid(event);
+        if (!uuid) return;
+
+        // Mark main icon as open or closed.
+        if (this._connectedIconsOpenClose[uuid]) delete this._connectedIconsOpenClose[uuid];
+        else this._connectedIconsOpenClose[uuid] = true;
+
+        // Trigger new icons to be shown or hidden.
+        this.render();
+    }
+
+    /**
+     * Add All equipped wireless items on the character to their PAN
+     * @param event
+     */
+    async _addAllEquippedWirelessDevicesToPAN(event) {
+        event.stopPropagation();
+        const matrixDevice = this.actor.getMatrixDevice();
+        if (matrixDevice) {
+            console.debug('Shadowrun5e | Adding all equipped wireless devices to actor PAN ->', event);
+            const progressBar = ui.notifications.info(game.i18n.localize("SR5.AddDevicesToPAN.Starting"), { progress: true });
```

Comment:
We _could_ also just add a `addSlaves` function to the network flow that would add all documents with one update. This would be just as quick as for a single device and make this progress bar (which is likely due to interface lag?) unnecessary.

Reply:
This turned out to not work very well as it was still waiting for all the devices sheets to render. I did remove the "disconnect" call so that items already on that PAN just get skipped over